src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/playback/GerritMissedEventsPlaybackManager.java

```diff
@@ -0,0 +1,486 @@
+        }
+        return restUrl + url;
+    }
+
+    /**
+     * Takes a timestamp and persists to xml file.
+     * @param evt Gerrit Event to persist.
+     * @return true if was able to persist event.
+     */
+    private synchronized boolean persist(GerritTriggeredEvent evt) {
+
+        long ts = evt.getEventCreatedOn().getTime();
+
+        if (ts == 0) {
+            logger.warn("Event CreatedOn is 0...Gerrit Server does not support attribute eventCreateOn. "
+                    + "Will NOT persist this event and Missed Events will be disabled!");
+            isSupported = false;
+            return false;
+        }
+
+        if (serverTimestamp != null && ts < serverTimestamp.getTimeSlice()) {
+            logger.debug("Event has same time slice {} or is earlier...NOT Updating time slice.", ts);
+        } else {
+            if (serverTimestamp == null) {
+                serverTimestamp = new EventTimeSlice(ts);
+                serverTimestamp.addEvent(evt);
+            } else {
+                if (ts > serverTimestamp.getTimeSlice()) {
+                    logger.debug("Current timestamp {} is GREATER than slice time {}.",
+                            ts, serverTimestamp.getTimeSlice());
+                    serverTimestamp = new EventTimeSlice(ts);
+                    serverTimestamp.addEvent(evt);
+                } else {
+                    if (ts == serverTimestamp.getTimeSlice()) {
+                        logger.debug("Current timestamp {} is EQUAL to slice time {}.",
+                                ts, serverTimestamp.getTimeSlice());
+                        serverTimestamp.addEvent(evt);
+                    }
+                }
+            }
+        }
+
+        XmlFile config = getConfigXml(serverName);
+        try {
+            config.write(serverTimestamp);
+        } catch (IOException e) {
+            logger.error(e.getMessage(), e);
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Shutdown the listener.
+     */
+    public void shutdown() {
+        GerritServer server = PluginImpl.getServer_(serverName);
+        if (server != null) {
+            server.removeListener((GerritEventListener)this);
+        } else {
+            logger.error("Could not find server {}", serverName);
+        }
+    }
+
+    /**
+     * @return whether playback is supported.
+     */
+    public boolean isSupported() {
+        return isSupported;
+    }
+
+    /**
+     * Return server timestamp.
+     * @return timestamp.
+     */
+    public EventTimeSlice getServerTimestamp() {
+        return serverTimestamp;
+    }
+
+    /**
+     * @param serverName The Name of the Gerrit Server to load config for.
+     * @return XmlFile corresponding to gerrit-trigger-server-timestamps.xml.
+     */
+    public static XmlFile getConfigXml(String serverName) {
+        File xmlFile = new File(Jenkins.getInstance().getRootDir() + GERRIT_SERVER_EVENT_DATA_FOLDER
```

Comment:
CheckForNull on ```Jenkins.getInstance()```

Try using the ```new File(parent, child)``` construct instead of string concatenation. Or ```jenkins.getRootPath().child(x).child(y)```.